### PR TITLE
feat(router): add auto retries configs in profile CRUD apis

### PR DESCRIPTION
**Pull request by sai-harsha-vardhan**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

add auto retries configs in profile CRUD apis
1. is_auto_retries_enabled - Boolean (denotes whether auto retries should be performed for a profile)
2. max_auto_retries_enabled - Int (maximum auto retries to be performed)

### Additional Changes

- [x] This PR modifies the API contract
- [x] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Tested Manually
1. Profile Create

```
curl --location '{{BASE_URL}}/account/{{MERCHANT_ID}}/business_profile' \
--header 'Content-Type: application/json' \
--header 'api-key: {{ADMIN_API_KEY}}' \
--data '{
    "is_auto_retries_enabled": true,
    "max_auto_retries_enabled": 1
}'
```

Response
`{
    "merchant_id": "merchant_1727371477",
    "profile_id": "pro_jIc12oz9zLreWO4gJqXZ",
    "profile_name": "default",
    "return_url": "https://example.com/success",
    "enable_payment_response_hash": true,
    "payment_response_hash_key": "uskD31l0vpyRXItGriQ2FZvyhgNJvY5eo3rpy7mnJ0iqm0ExGMgenVstuaVaTxtI",
    "redirect_to_merchant_with_http_post": false,
    "webhook_details": {
        "webhook_version": "1.0.1",
        "webhook_username": "ekart_retail",
        "webhook_password": "password_ekart@123",
        "webhook_url": "https://eop67yav2axa2pv.m.pipedream.net/",
        "payment_created_enabled": true,
        "payment_succeeded_enabled": true,
        "payment_failed_enabled": true
    },
    "metadata": null,
    "routing_algorithm": null,
    "intent_fulfillment_time": 900,
    "frm_routing_algorithm": null,
    "payout_routing_algorithm": null,
    "applepay_verified_domains": null,
    "session_expiry": 900,
    "payment_link_config": null,
    "authentication_connector_details": null,
    "use_billing_as_payment_method_billing": true,
    "extended_card_info_config": null,
    "collect_shipping_details_from_wallet_connector": false,
    "collect_billing_details_from_wallet_connector": false,
    "always_collect_shipping_details_from_wallet_connector": false,
    "always_collect_billing_details_from_wallet_connector": false,
    "is_connector_agnostic_mit_enabled": false,
    "payout_link_config": null,
    "outgoing_webhook_custom_http_headers": null,
    "tax_connector_id": null,
    "is_tax_connector_enabled": false,
    "is_network_tokenization_enabled": false,
    "is_auto_retries_enabled": true,
    "max_auto_retries_enabled": 1
}`

2. Profile Update

```
curl --location '{{BASE_URL}}/account/{{MERCHANT_ID}}/business_profile/{{PROFILE_ID}}' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: {{ADMIN_API_KEY}}' \
--data '{
    "is_auto_retries_enabled": false,
    "max_auto_retries_enabled": 2
}'
```

Response:
`{
    "merchant_id": "merchant_1727371477",
    "profile_id": "pro_jIc12oz9zLreWO4gJqXZ",
    "profile_name": "default",
    "return_url": "https://example.com/success",
    "enable_payment_response_hash": true,
    "payment_response_hash_key": "uskD31l0vpyRXItGriQ2FZvyhgNJvY5eo3rpy7mnJ0iqm0ExGMgenVstuaVaTxtI",
    "redirect_to_merchant_with_http_post": false,
    "webhook_details": {
        "webhook_version": "1.0.1",
        "webhook_username": "ekart_retail",
        "webhook_password": "password_ekart@123",
        "webhook_url": "https://eop67yav2axa2pv.m.pipedream.net/",
        "payment_created_enabled": true,
        "payment_succeeded_enabled": true,
        "payment_failed_enabled": true
    },
    "metadata": null,
    "routing_algorithm": null,
    "intent_fulfillment_time": 900,
    "frm_routing_algorithm": null,
    "payout_routing_algorithm": null,
    "applepay_verified_domains": null,
    "session_expiry": 900,
    "payment_link_config": null,
    "authentication_connector_details": null,
    "use_billing_as_payment_method_billing": true,
    "extended_card_info_config": null,
    "collect_shipping_details_from_wallet_connector": false,
    "collect_billing_details_from_wallet_connector": false,
    "always_collect_shipping_details_from_wallet_connector": false,
    "always_collect_billing_details_from_wallet_connector": false,
    "is_connector_agnostic_mit_enabled": false,
    "payout_link_config": null,
    "outgoing_webhook_custom_http_headers": null,
    "tax_connector_id": null,
    "is_tax_connector_enabled": false,
    "is_network_tokenization_enabled": false,
    "is_auto_retries_enabled": false,
    "max_auto_retries_enabled": 2
}`

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by jarnura — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

File: crates/api_models/src/admin.rs [hunk 1]
```diff
@@ -2202,6 +2208,13 @@ pub struct ProfileResponse {
     /// If set to `true` is_network_tokenization_enabled will be checked.
     #[schema(default = false, example = false)]
     pub is_network_tokenization_enabled: bool,
+
+    /// Indicates if is_auto_retries_enabled is enabled or not.
+    #[schema(default = false, example = false)]
```
Comment (Narayanbhat166) [inline crates/api_models/src/admin.rs:2213]: I don't think default has any significance when seriallizing.

**Review by Narayanbhat166 — APPROVED** [review]

File: crates/api_models/src/admin.rs [hunk 2]
```diff
@@ -1971,6 +1971,12 @@ pub struct ProfileCreate {
     /// If set to `true` is_network_tokenization_enabled will be checked.
     #[serde(default)]
     pub is_network_tokenization_enabled: bool,
+
+    /// Indicates if is_auto_retries_enabled is enabled or not.
```
Comment (Narayanbhat166) [inline crates/api_models/src/admin.rs:1975]: we can provide a default value here, that we are inferring in domain model.

File: crates/api_models/src/admin.rs [hunk 3]
```diff
@@ -1971,6 +1971,12 @@ pub struct ProfileCreate {
...
+    pub is_auto_retries_enabled: Option<bool>,
+
+    /// Maximum number of auto retries allowed for a payment
```
Comment (Narayanbhat166) [inline crates/api_models/src/admin.rs:1978]: can have max value in the documentation

Comment (sai-harsha-vardhan) [reply crates/api_models/src/admin.rs:1978]: Sure, will make this changes in my next PR

**Pull request merged**